### PR TITLE
ARM64: GenerateFastShiftRight used use LDIMM if src is an immediate

### PR DESCRIPTION
**Pull request by agarwal-sandeep**

### CODE REVIEW

File: lib/Backend/arm64/LowerMD.cpp [hunk 1]
```diff
@@ -3710,9 +3710,20 @@ LowererMD::GenerateFastShiftRight(IR::Instr * instrShift)
...
+            opndSrc2 = IR::IntConstOpnd::New(Js::TaggedInt::ToInt32(opndSrc2->AsAddrOpnd()->m_address), TyInt32, this->m_func);
+            instr = IR::Instr::New(Js::OpCode::LDIMM, opndReg, opndSrc2, this->m_func);
+        }
+        else
+        {
+            opndSrc2 = opndSrc2->UseWithNewType(TyInt32, this->m_func);
+            instr = IR::Instr::New(Js::OpCode::MOV, opndReg, opndSrc2, this->m_func);
+        }
+
         instrShift->InsertBefore(instr);
```
Comment (sigatrev) [inline lib/Backend/arm64/LowerMD.cpp]: You should be able to use Lowerer::InsertMove here

Comment (agarwal-sandeep) [reply lib/Backend/arm64/LowerMD.cpp]: Yup and at other places too. Changed.

---
In reply to: [150096843](https://github.com/Microsoft/ChakraCore/pull/4189#discussion_r150096843) [](ancestors = 150096843)

**Comment by rajatd** [comment]
>     instr = IR::Instr::New(Js::OpCode::MOV, instrShift->GetDst(), opndReg, this->m_func);

change this to InsertMove as well.

---
Refers to: lib/Backend/arm64/LowerMD.cpp:3734 in 5837b8f. [](commit_id = 5837b8f62a5c407e49c4dbe275410876c531d847, deletion_comment = False)

**Comment by agarwal-sandeep** [comment]
>     instr = IR::Instr::New(Js::OpCode::MOV, instrShift->GetDst(), opndReg, this->m_func);

opndReg should be a reg here so MOV is ok

---
In reply to: [343340568](https://github.com/Microsoft/ChakraCore/pull/4189#issuecomment-343340568) [](ancestors = 343340568)

---
Refers to: lib/Backend/arm64/LowerMD.cpp:3734 in 5837b8f. [](commit_id = 5837b8f62a5c407e49c4dbe275410876c531d847, deletion_comment = False)

**Pull request merged**